bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -148,7 +154,7 @@ void ReplicaImp::messageHandler(MessageBase *msg) {
       }
     }
   }
-  if (!isCollectingState() && validateMessage(trueTypeObj)) {
+  if (!isCollectingState()) {
```

Comment:
validateMessage() should be called here.

Reply:
Again? Its already being called at line 131.